src/index.ts

```diff
@@ -44,32 +45,13 @@ async function run() {
       await exec.exec(`git config --global user.email "${githubUserEmail}"`);
     }
 
-    await exec.exec(`git fetch origin ${headRef} ${baseRef}`);
+    await exec.exec(`git fetch origin ${headRef} ${baseRef} ${defaultBranch}`);
     
     await exec.exec(`git checkout ${baseRef}`);
 
-    console.log(`\n\n Merging ${headRef} into ${baseRef} using '--strategy-option theirs'`);
+    console.log(`\n\n Merging ${headRef} into ${baseRef}`);
 
-    await exec.exec(`git merge origin/${headRef} --allow-unrelated-histories --strategy-option theirs`);
-
-    console.log(`\n\n Checking for deleted files in ${headRef}`);
-
-    let conflictFiles = '';
-    const gitDiffListeners = {
-      stdout: (data: Buffer) => {
-        conflictFiles += data.toString();
-      }
-    };
-
-    await exec.exec(`git diff --name-status --diff-filter=A origin/${headRef}`, undefined, { listeners: gitDiffListeners });
-
-    const filesSeparator = new RegExp(/A\t/g);
-    console.log(`\n\nFiles that was removed in ${headRef}, but still exist in ${baseRef}:\n${conflictFiles.replace(filesSeparator, '')}`);
-    if (conflictFiles.trim().length != 0) {
-      console.log(`\n\nRemoving conflicted files from ${baseRef}`);
-      await exec.exec(`git rm ${conflictFiles.replace(filesSeparator, ' ').replace(/\n/g, '')}`);
-      await exec.exec(`git commit --amend --no-edit`);
-    }
+    await exec.exec(`git merge origin/${headRef} --allow-unrelated-histories`);
```

Comment:
do we need to keep `--allow-unrelated-histories` tho, seems like branches should always be synchronised now?

Reply:
Same for another place

Reply:
`--allow-unrelated-histories` won't do anything in case if everything is good, but can save us from potential failing otherwise

Reply:
`headRef`, `baseRef` and `defaultBranch` are called so since they can be different for different projects (they aren't now) and so they have a more dynamic names. `headRef` and `baseRef` can be anything depending on the PR head and base and `defaultBranch` can be set as a parameter (`dev` is default). They surely can be renamed if you think we do not need any flexibility here

Reply:
I would remove `--allow-unrelated-histories` and rather fail explicitly.
In updated flow we don't expect the case with unrelated histories to ever happen, and if it does - it's better to fail on workflow level, IMO

In regards to branches - I personally don't think in this place we need to be flexible, but up to you